### ISSUE TITLE
No results, always exit code 1

### ISSUE THREAD

**Issue by cfjedimaster** [sonarwhal/sonar#327]
I've tried to sonar 4 sites now and every single time I get exit code 1. I added debug and get this:

sonar:engine Starting the analysis on / +2ms
  sonar:collectors:cdp:cdp-launcher Error reading /home/ray/cdp.pid +7ms
  sonar:collectors:cdp:cdp-launcher Error: ENOENT: no such file or directory, open '/home/ray/cdp.pid'
  sonar:collectors:cdp:cdp-launcher     at Object.fs.openSync (fs.js:638:18)
  sonar:collectors:cdp:cdp-launcher     at Object.fs.readFileSync (fs.js:540:33)
  sonar:collectors:cdp:cdp-launcher     at getPid (/usr/local/lib/node_modules/@sonarwhal/sonar/dist/src/lib/collectors/cdp/cdp-launcher.js:149:27)
  sonar:collectors:cdp:cdp-launcher     at launchChrome (/usr/local/lib/node_modules/@sonarwhal/sonar/dist/src/lib/collectors/cdp/cdp-launcher.js:195:24)
  sonar:collectors:cdp:cdp-launcher     at <anonymous> +0ms
  sonar:collectors:cdp:cdp-launcher Trying to open Chrome on Linux +14ms
  sonar:collectors:cdp:cdp-launcher Executing undefined +84ms
  sonar:collectors:cdp:cdp-launcher Error executing command +2ms
  sonar:collectors:cdp:cdp-launcher TypeError: "file" argument must be a non-empty string
  sonar:collectors:cdp:cdp-launcher     at normalizeSpawnArguments (child_process.js:360:11)
  sonar:collectors:cdp:cdp-launcher     at Object.exports.spawn (child_process.js:465:38)
  sonar:collectors:cdp:cdp-launcher     at launchChrome (/usr/local/lib/node_modules/@sonarwhal/sonar/dist/src/lib/collectors/cdp/cdp-launcher.js:235:39)
  sonar:collectors:cdp:cdp-launcher     at <anonymous> +2ms
  sonar:collectors:cdp Error connecting to browser +26ms
  sonar:collectors:cdp TypeError: "file" argument must be a non-empty string
  sonar:collectors:cdp     at normalizeSpawnArguments (child_process.js:360:11)
  sonar:collectors:cdp     at Object.exports.spawn (child_process.js:465:38)
  sonar:collectors:cdp     at launchChrome (/usr/local/lib/node_modules/@sonarwhal/sonar/dist/src/lib/collectors/cdp/cdp-launcher.js:235:39)
  sonar:collectors:cdp     at <anonymous> +0ms
  sonar:cli Failed to analyze: https://github.com/ +4ms
  sonar:collectors:cdp Closing browsers used by CDP +1ms
  sonar:collectors:cdp Couldn't close the client properly +0ms

**Comment by alrra** [comment]
@cfjedimaster Thanks for taking the time to report this!

---

> Error reading /home/ray/cdp.pid +7ms

@sarvaje Can you take a look into it?

**Comment by molant** [comment]
This is probably something with permissions and related to the previous problem @cfjedimaster had. Maybe solved via #311 once it is done?

**Comment by sarvaje** [comment]
I'll take a look to this.

**Comment by sarvaje** [comment]
@alrra I don't think the problem is related with
>Error reading /home/ray/cdp.pid +7ms

It looks like a problem launching chrome, let see if it is fixed in #331. In that PR we are using the package `chrome-launcher`

**Comment by sarvaje** [comment]
@cfjedimaster we have just release a [new version](https://github.com/sonarwhal/sonar/releases/tag/0.1.0) with a new launcher for chrome. Can you try it an let me know if the problem continue or not?

Thanks!

**Comment by cfjedimaster** [comment]
No change. Should I `sonar init` again perhaps - or clean out some cache?

**Comment by cfjedimaster** [comment]
Oh, it looks to be a new error - correction - not new - but I think I'm seeing more stuff. Either way...

```
sonar:engine Rules loaded: 14 +23ms
  sonar:engine Starting the analysis on / +29ms
  sonar:collectors:shared:launcher Error reading /home/ray/cdp.pid +46ms
  sonar:collectors:shared:launcher Error: ENOENT: no such file or directory, open '/home/ray/cdp.pid' +3ms
  sonar:collectors:shared:launcher Error launching browser +410ms
  sonar:collectors:shared:launcher Error: The environment variable LIGHTHOUSE_CHROMIUM_PATH must be set to executable of a build of Chromium version 54.0 or la
ter.
  sonar:collectors:shared:launcher     at Object.linux (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-finder.js:83:15)
  sonar:collectors:shared:launcher     at Launcher.<anonymous> (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:10
7:75)
  sonar:collectors:shared:launcher     at Generator.next (<anonymous>)
  sonar:collectors:shared:launcher     at /usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:12:71
  sonar:collectors:shared:launcher     at Promise (<anonymous>)
  sonar:collectors:shared:launcher     at __awaiter (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:8:12)
  sonar:collectors:shared:launcher     at Launcher.launch (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:92:16)
  sonar:collectors:shared:launcher     at Object.<anonymous> (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:42:2
4)
  sonar:collectors:shared:launcher     at Generator.next (<anonymous>)
  sonar:collectors:shared:launcher     at /usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:12:71 +0ms
  sonar:collectors:shared:remote-debugging-collector Error connecting to browser +4ms
  sonar:collectors:shared:remote-debugging-collector Error: The environment variable LIGHTHOUSE_CHROMIUM_PATH must be set to executable of a build of Chromium
version 54.0 or later.
  sonar:collectors:shared:remote-debugging-collector     at Object.linux (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-find
er.js:83:15)
  sonar:collectors:shared:remote-debugging-collector     at Launcher.<anonymous> (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chr
ome-launcher.js:107:75)
  sonar:collectors:shared:remote-debugging-collector     at Generator.next (<anonymous>)
  sonar:collectors:shared:remote-debugging-collector     at /usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:12:71
  sonar:collectors:shared:remote-debugging-collector     at Promise (<anonymous>)
  sonar:collectors:shared:remote-debugging-collector     at __awaiter (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launche
r.js:8:12)
  sonar:collectors:shared:remote-debugging-collector     at Launcher.launch (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-l
auncher.js:92:16)
  sonar:collectors:shared:remote-debugging-collector     at Object.<anonymous> (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrom
e-launcher.js:42:24)
  sonar:collectors:shared:remote-debugging-collector     at Generator.next (<anonymous>)
  sonar:collectors:shared:remote-debugging-collector     at /usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:12:71
+0ms
  sonar:cli Failed to analyze: https://www.raymondcamden.com/ +9ms
  sonar:cli Error: The environment variable LIGHTHOUSE_CHROMIUM_PATH must be set to executable of a build of Chromium version 54.0 or later.
  sonar:cli     at Object.linux (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-finder.js:83:15)
  sonar:cli     at Launcher.<anonymous> (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:107:75)
  sonar:cli     at Generator.next (<anonymous>)
  sonar:cli     at /usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:12:71
  sonar:cli     at Promise (<anonymous>)
  sonar:cli     at __awaiter (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:8:12)
  sonar:cli     at Launcher.launch (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:92:16)
  sonar:cli     at Object.<anonymous> (/usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:42:24)
  sonar:cli     at Generator.next (<anonymous>)
  sonar:cli     at /usr/local/lib/node_modules/@sonarwhal/sonar/node_modules/chrome-launcher/chrome-launcher.js:12:71 +0ms
  sonar:collectors:shared:remote-debugging-collector Closing browsers used by CDP +5ms
  sonar:collectors:shared:remote-debugging-collector Couldn't close the client properly +3ms
  sonar:cli Total runtime: 483ms +3ms
```

**Comment by sarvaje** [comment]
It is something related to launch chrome, with os are you using and chrome version?

**Comment by molant** [comment]
@sarvaje why do we have this error message?

sonar:cli Error: The environment variable LIGHTHOUSE_CHROMIUM_PATH must be set to executable of a build of Chromium version 54.0 or later.

It doesn't make too much sense to be using that variable.

**Comment by sarvaje** [comment]
`chrome-launcher` look for that variable

**Comment by molant** [comment]
But does it also set it or how does it work exactly?

**Comment by sarvaje** [comment]
`chrome-launcher` look for all the chrome installations you have in your machine, and looks like you can set the path of the instalation in that variable
```
if (canAccess(process.env.LIGHTHOUSE_CHROMIUM_PATH)) {
    installations.push(process.env.LIGHTHOUSE_CHROMIUM_PATH);
}
```
I'm not setting it, it is something the package support

**Comment by cfjedimaster** [comment]
I'm on Windows 10, using WSL though.

**Comment by sarvaje** [comment]
> I'm on Windows 10, using WSL though.

I think that is the problem:
On linux, the launcher is looking for chrome in these folders
```
const desktopInstallationFolders = [
    path.join(require('os').homedir(), '.local/share/applications/'),
    '/usr/share/applications/',
];
```

**Comment by cfjedimaster** [comment]
I'll switch to Powershell. If this isn't supported (and that would be understandable), then it could be just a doc issue. :)

**Comment by cfjedimaster** [comment]
Working there - although I hate all the errors it found. :)

**Comment by molant** [comment]
@cfjedimaster you can use sonar with WSL if you use the jsdom collector instead of CDP.
We will update the documentation.

Thanks!!

**Comment by cfjedimaster** [comment]
Confirmed. I saw a new bug that seems minor and will file it in a new issue.

**Comment by molant** [comment]
This seems to be solved then. @cfjedimaster reopen if this is not the case.

Thanks!

**Comment by cfjedimaster** [comment]
Just an FYI, I've been away from Sonar for a while. Reinstalled on WSL and ran into an issue. I had not selected jsdon. Looking here (https://sonarwhal.com/docs/user-guide/) I do not see a warning about WSL in regards to using jsdom.

**Comment by molant** [comment]
WSL should work with jsdom but not chrome.
What site, node version and sonar version are you using? What is your sonarrc file?

The new issue template has some fields to help us debug better. Thanks!

**Comment by cfjedimaster** [comment]
To be clear, I'm not saying jsdom doesn't work - what I meant is - the docs weren't updated to warn people to use jsdom. (Specifically the guide I linked to.)

**Comment by alrra** [comment]
> To be clear, I'm not saying jsdom doesn't work - what I meant is - the docs weren't updated to warn people to use jsdom. (Specifically the guide I linked to.)

@cfjedimaster I agree, we should specify that in the docs. Care to open a pull request?

Also, in the future, we should be better at what we recommend, so I've opened #517.

**Comment by cfjedimaster** [comment]
PR #523 added.
